packages/data-context/src/data/ProjectConfigManager.ts

```diff
@@ -311,7 +311,8 @@ export class ProjectConfigManager {
   }
 
   onLoadError = (error: any) => {
-    this.closeWatchers()
+    // Tyler TODO: Do we need to close here? Making this async has downstream effects
```

Comment:
@tbiethman any idea about this?

Reply:
My comment is related to our calling this in the ProjectLifecycleManager (https://github.com/cypress-io/cypress/blob/develop/packages/data-context/src/data/ProjectLifecycleManager.ts#L802); making this async would have ramifications to a number of APIs there without some refactoring.

It might be necessary though. Keeping the watcher open while in the error state would result in `refreshLifecycle` being [called on change](https://github.com/cypress-io/cypress/pull/22606/files#diff-d81c2004d6fb4e9796dcb278f97ccf435abb59ddbb4b41d8f9706bf5f26f8af7R338), which wouldn't be expected in the previous code. We could try forcing an error and seeing what impact this has.

Reply:
I'm attempting some refactoring here to allow us to keep closing the watchers here. The good news is we have lots of tests around this, the bad news is those tests are now failing 😞